File: src/test/java/com/sy/coladay/error/BaseErrorControllerTest.java
package com.sy.coladay.error;

import static org.mockito.Mockito.verify;

import javax.servlet.http.HttpServletResponse;
import lombok.SneakyThrows;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;
import org.springframework.http.HttpStatus;

/**
 * Test class for {@link BaseErrorController}.
 */
@ExtendWith(MockitoExtension.class)
class BaseErrorControllerTest {

  @Mock
  HttpServletResponse response;

  @InjectMocks
  BaseErrorController baseErrorController;

  @Test
  @SneakyThrows
  void handleDataIntegrityViolationException_return_409() {

    baseErrorController.handleDataIntegrityViolationException(response);
    verify(response).sendError(HttpStatus.CONFLICT.value(),
        "The data presented are in conflict with existing ones");
  }
}

File: src/test/kotlin/com/sy/coladay/error/BaseErrorControllerTest.kt
package com.sy.coladay.error

import lombok.SneakyThrows
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.junit.jupiter.MockitoExtension
import org.springframework.http.HttpStatus
import javax.servlet.http.HttpServletResponse

/**
 * Test class for [BaseErrorController].
 */
@ExtendWith(MockitoExtension::class)
internal class BaseErrorControllerTest {

    @Mock
    var response: HttpServletResponse? = null

    @InjectMocks
    var baseErrorController: BaseErrorController? = null

    @Test
    @SneakyThrows
    fun `returns a 409 when a data integrity violation exception is raised`() {
        baseErrorController!!.handleDataIntegrityViolationException(response!!)
        verify(response)?.sendError(
            HttpStatus.CONFLICT.value(),
            "The data presented are in conflict with existing ones"
        )
    }
}